### PR TITLE
[ANIMATION WORKLET] Calculate viewport-margins for scroll linked animation

### PR DESCRIPTION
**Pull request by nainar**
This also passes along the values for the `intersection-ratio`s however, those aren't used. Can split that up into a separate PR if needed.

### CODE REVIEW

File: extensions/amp-animation/0.1/amp-animation.js [hunk 1]
```diff
@@ -291,9 +291,12 @@ export class AmpAnimation extends AMP.BaseElement {
     // The animation will be triggered (in paused state) and seek will happen
     // regardless of visibility
     this.triggered_ = true;
-    this.hasPositionObserver_ = !!invocation.caller &&
-      invocation.caller.tagName === 'AMP-POSITION-OBSERVER';
-    return this.createRunnerIfNeeded_().then(() => {
+    this.positionObserver_ = invocation.caller;
+    let positionObserverArgs = null;
+    if (this.positionObserver_) {
+      positionObserverArgs = this.parseAttributes_(this.positionObserver_);
```
Comment (aghassemi) [inline extensions/amp-animation/0.1/amp-animation.js]: By the time position-observer send the seekTo event, it should have already resolved and calculated all these values, you should be able to send computed values as part of the event and access it here. That way `parseAttributes_`, `applyMargins_` and `validateAndResolveMargin_` can be removed.

amp-animation should not have knowledge of amp-position-observer attributes. I think all the coupling should be through the event channel.

Comment (nainar) [reply extensions/amp-animation/0.1/amp-animation.js]: Done as per offline conversation added a datastructure on Event in `amp-position-observer` which I then access in `amp-animation`

File: extensions/amp-animation/0.1/amp-animation.js [hunk 2]
```diff
@@ -408,12 +410,14 @@ export class AmpAnimation extends AMP.BaseElement {
   /**
    * Creates the runner but animations will not start.
    * @param {?JsonObject=} opt_args
+   * @param {?Object=} opt_positionObserverArgs
    * @return {!Promise}
    * @private
    */
-  createRunnerIfNeeded_(opt_args) {
+  createRunnerIfNeeded_(opt_args, opt_positionObserverArgs) {
```
Comment (aghassemi) [inline extensions/amp-animation/0.1/amp-animation.js]: `opt_viewportData` is cleaner IMO. Avoiding position-observer naming and logic throughout the amp-animation stack would be nice.

Comment (nainar) [reply extensions/amp-animation/0.1/amp-animation.js]: Done.

**Pull request merged**